elpis/engines/hft/objects/transcription.py

```diff
@@ -0,0 +1,257 @@
+        with torch.no_grad():
+            logits = model(input_values).logits
+        predicted_ids = torch.argmax(logits, dim=-1)
+        self._set_stage(PROCESS_INPUT, complete=True)
+
+        # transcribe
+        self._set_stage(TRANSCRIPTION)
+        transcription = processor.decode(predicted_ids[0])
+        print('transcription', transcription)
+        self._set_stage(TRANSCRIPTION, complete=True)
+
+        self._set_stage(SAVING)
+        print('=== Save transcription')
+        self._save_transcription(transcription)
+
+        self._set_stage(SAVING, msg='Saved transcription, generating utterances')
+        # Utterances to be used creating elan files
+        print('=== Generate utterances')
+        utterances = self._generate_utterances(
+            processor, predicted_ids, input_values, transcription)
+        print('=== Save utterances (elan and text)')
+        self._save_utterances(utterances)
+
+        self._set_stage(SAVING, complete=True)
+        self._set_finished_transcription(True)
+        if on_complete is not None:
+            on_complete()
+
+    def text(self):
+        with open(self.text_path, 'r') as fin:
+            text = fin.read()
+            return text
+
+    def elan(self):
+        with open(self.elan_path, 'r') as fin:
+            return fin.read()
+
+    def get_confidence(self):
+        return None
+
+    def _get_wav2vec2_requirements(self) -> Tuple[Wav2Vec2Processor, Wav2Vec2ForCTC]:
+        """Builds and returns pretrained Wav2Vec2 Processor and Model from the
+        project path.
+        """
+        pretrained_path = Path(self.model.path) / "wav2vec2"
+
+        processor = Wav2Vec2Processor.from_pretrained(pretrained_path)
+        model = Wav2Vec2ForCTC.from_pretrained(pretrained_path)
+
+        return processor, model
+
+    def _generate_utterances(self,
+                            processor: Wav2Vec2Processor,
+                            predicted_ids: torch.Tensor,
+                            input_values: torch.Tensor,
+                            transcription: str) -> Tuple[List[str], List[float], List[float]]:
+        """Generates a mapping of words to their start and end times from a transcription.
+
+        Parameters:
+            processor: The wav2vec2 processor from which we take the tokenizer.
+            predicted_ids 
+        """
+        words = [word for word in transcription.split(' ') if len(word) > 0]
+        predicted_ids = predicted_ids[0].tolist()
+
+        # Determine original sample rate
+        original_file = Path(f'/tmp/{self.hash}/original.wav')
+        _, sample_rate = self._load_audio(original_file)
+
+        # Add times to ids
+        duration_sec = input_values.shape[1] / sample_rate
+        print('Audio length:', duration_sec)
+
+        time_from_index = lambda index: index / len(predicted_ids) * duration_sec
+        generate_timestamps = lambda item: (time_from_index(item[0]), item[1])
```

Comment:
So make these underscore methods if they're only going to be used once? I think my thought process was that I was trying not to clutter up the methods defined on this class, and thought since these were all local to this method that I could chuck them here, what do you think?

Reply:
They're either reused enough to justify a function or they don't really need a name. This isn't a huge deal either way, it just seems superfluous to give them a variable name for a single use when a comment and usage in place would do.